engines/illusions/actor.cpp

```diff
@@ -0,0 +1,1546 @@
+			Font *font = _vm->findFont(_fontId);
+			_surface->fillRect(Common::Rect(surfInfo._dimensions._width, surfInfo._dimensions._height), 0);
+			gfx_sub_40CA70(_surface, font, _field18C, _surfInfo._dimensions, _field198);
+			*/
+			_flags |= Illusions::ACTOR_FLAG_4000;
+		}
+		else {
+			_flags |= Illusions::ACTOR_FLAG_2000;
+			_flags |= Illusions::ACTOR_FLAG_4000;
+		}
+	}
+}
+
+void Actor::destroySurface() {
+	if (_surface) {
+		_surface->free();
+		delete _surface;
+		_surface = 0;
+	}
+}
+
+void Actor::initSequenceStack() {
+	_seqStackCount = 0;
+}
+
+void Actor::pushSequenceStack(int16 value) {
+	_seqStack[_seqStackCount++] = value;
+}
+
+int16 Actor::popSequenceStack() {
+	return _seqStack[--_seqStackCount];
+}
+
+void Actor::setControlRoutine(ActorControlRoutine *controlRoutine) {
+	delete _controlRoutine;
+	_controlRoutine = controlRoutine;
+}
+
+void Actor::runControlRoutine(Control *control, uint32 deltaTime) {
+	if (_controlRoutine)
+		(*_controlRoutine)(control, deltaTime);
+}
+
+bool Actor::findNamedPoint(uint32 namedPointId, Common::Point &pt) {
+	if (_namedPoints->findNamedPoint(namedPointId, pt)) {
+		pt.x += _position.x;
+		pt.y += _position.y;
+		return true;
+	}
+	return false;
+}
+
+// Control
+
+Control::Control(IllusionsEngine *vm)
+	: _vm(vm) {
+	_flags = 0;
+	_pauseCtr = 0;
+	_priority = 0;
+	_objectId = 0;
+	_unkPt.x = 0;
+	_unkPt.y = 0;
+	_pt.x = 0;
+	_pt.y = 0;
+	_feetPt.x = 0;
+	_feetPt.y = 0;
+	_position.x = 0;
+	_position.y = 0;
+	_actorTypeId = 0;
+	_actor = 0;
+	_sceneId = _vm->getCurrentScene();
+}
+
+Control::~Control() {
+}
+
+void Control::pause() {
+
+	_vm->_dict->setObjectControl(_objectId, 0);
+
+	if (_objectId == Illusions::CURSOR_OBJECT_ID)
+		_vm->setCursorControl(0);
+
+	if (_actor && !(_actor->_flags & Illusions::ACTOR_FLAG_200))
+		_actor->destroySurface();
+
+}
+
+void Control::unpause() {
+
+	_vm->_dict->setObjectControl(_objectId, this);
+
+	if (_objectId == Illusions::CURSOR_OBJECT_ID)
+		_vm->setCursorControl(this);
+
+	if (_actor && !(_actor->_flags & Illusions::ACTOR_FLAG_200)) {
+		SurfInfo surfInfo;
+		ActorType *actorType = _vm->_dict->findActorType(_actorTypeId);
+		if (actorType)
+			surfInfo = actorType->_surfInfo;
+		else
+			surfInfo = _actor->_surfInfo;
+		_actor->createSurface(surfInfo);
+	}
+
+}
+
+void Control::appearActor() {
+	if (_vm->getGameId() == kGameIdDuckman) {
+		_flags |= 1;
+		_actor->_flags |= Illusions::ACTOR_FLAG_IS_VISIBLE;
+		if (_objectId == Illusions::CURSOR_OBJECT_ID) {
+			if (_actor->_frameIndex) {
+				_actor->_flags |= Illusions::ACTOR_FLAG_2000;
+				_actor->_flags |= Illusions::ACTOR_FLAG_4000;
+			}
+			_vm->_input->discardAllEvents();
+		}
+	} else {
+		if (_objectId == Illusions::CURSOR_OBJECT_ID) {
+			_vm->showCursor();
+		} else {
+			if (_actor->_frameIndex || _actorTypeId == 0x50004)
+				_actor->_flags |= Illusions::ACTOR_FLAG_IS_VISIBLE;
+			else
+				_actor->_flags |= Illusions::ACTOR_FLAG_1000;
+			for (uint i = 0; i < kSubObjectsCount; ++i)
```

Comment:
Since the loop body isn't a single line the `for` should probably come with braces to enhance readability.